crates/ruff_linter/src/checkers/ast/analyze/statement.rs

```diff
@@ -1131,6 +1131,9 @@ pub(crate) fn statement(stmt: &Stmt, checker: &mut Checker) {
             if checker.enabled(Rule::TooManyNestedBlocks) {
                 pylint::rules::too_many_nested_blocks(checker, stmt);
             }
+            if checker.enabled(Rule::UnnecessaryAssignment) {
+                pylint::rules::unnecessary_assignment(checker, if_);
+            }
```

Comment:
I applied this rule by checking whether the previous statement from a `IfStmt` is an `AssignStmt`, however I am wondering if it is more desirable to do check if the next statement of a `AssignStmt` is an `IfStmt`?

Reply:
I agree that assignments are far more common than if statements.

I think the answer depends on the costs for checking `previous_statement` and `next_statement`.

My thought here is that retrieving the previous statement using the newly added `previous_statement` might be more expensive because it has to iterate over all previous statements using `previous_statements` to find the previous statement (if there is a better way, please let me know).

While checking an assignment, then checking the `next_statement` might be a cheap operation.

Do you have any idea? If they have equal cost I think the current implementation is fine. 😄